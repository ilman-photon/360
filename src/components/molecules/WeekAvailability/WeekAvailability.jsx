import Box from "@mui/material/Box";
import React from "react";
import Link from "@mui/material/Link";
import { StyledButton } from "../../atoms/Button/button";
import styles from "./styles.module.scss";
import KeyboardArrowDownOutlinedIcon from "@mui/icons-material/KeyboardArrowDownOutlined";
import { Divider, Typography } from "@mui/material";
import { useRouter } from "next/router";

export function viewAllAvailabilityLinkUI({
  onClickViewAllAvailability = () => {
    // This is intentional
  },
}) {
  return (
    <Box
      sx={{
        gridArea: "linkAvability",
        display: "flex",
        alignItems: "center",
        justifyContent: "center",
        p: 3,
      }}
      className={styles.linkWrapper}
    >
      <Link
        className={styles.linkAvailabelity}
        onClick={onClickViewAllAvailability}
      >
        View all Availability
      </Link>
    </Box>
  );
}

export const WeekAvailability = ({
  scheduleData = {
    monday: ["11:30am", "", "", ""],
    tuesday: ["08:00am", "10:30am", "11:00am", "3 more"],
    wednesday: ["08:30am", "10:30am", "11:30am", "5 more"],
    thursday: ["09:30am", "11:00am", "", ""],
    friday: ["09:30am", "", "", ""],
    saturday: ["09:30am", "", "", ""],
  },
  onClickViewAllAvailability = () => {
    // This is intentional
  },
  keyWeek = "",
}) => {
  const router = useRouter();
  function renderScheduleData() {
    let renderUI = [];
    for (const [key, value] of Object.entries(scheduleData)) {
      for (let i = 0; i < value.length; i++) {
        let gridArea = `${key}Schedule${i}`;
        let isTypeMore = false;
        if (value[i].indexOf("more") > -1) {
          gridArea = `more${key}Schedule`;
          isTypeMore = true;
        }
        renderUI.push(
          buttonSchedule(
            value[i],
            gridArea,
            isTypeMore,
            `${keyWeek}-${i}-${key}-schedule-button`
          )
        );
      }
    }
    return renderUI;
  }

  function buttonSchedule(label, gridArea, isTypeMore = false, index = "") {
    if (label) {
      return (
        <Box
          key={index}
          sx={{ gridArea: gridArea, width: "90px" }}
          className={styles.buttonWrapper}
        >
          <StyledButton
            theme={constants.PATIENT}
            mode={constants.PRIMARY}
            type="submit"
            size={constants.SMALL}
            gradient={false}
            className={styles.scheduleBtn}
            onClick={() => {
              //TO DO: temporary navigate, move to page when start developing functionality
              router.push("/patient/schedule-appointment");
            }}
          >
            <Typography className={styles.scheduleBtnLabel}>
              {label}
              {isTypeMore && (
                <KeyboardArrowDownOutlinedIcon sx={{ width: "18px" }} />
              )}
            </Typography>
          </StyledButton>
        </Box>
      );
    }
    return (
      <Box key={index} className={styles.dividerContainer}>
        <Divider className={styles.divider} />
      </Box>
    );
  }

  const constants = require("../../../utils/constants");
  return (
    <Box
      key={keyWeek}
      sx={{
        display: "grid",
        width: 598,
        gridTemplateColumns: "repeat(6, 1fr)",
        gap: 1,
        justifyContent: "center",
        alignContent: "center",
        gridTemplateRows: "auto",
        gridTemplateAreas: `"mondaySchedule0 tuesdaySchedule0 wednesdaySchedule0 thursdaySchedule0 fridaySchedule0 saturdaySchedule0"
                            "mondaySchedule1 tuesdaySchedule1 wednesdaySchedule1 thursdaySchedule1 fridaySchedule1 saturdaySchedule1"
                            "mondaySchedule2 tuesdaySchedule2 wednesdaySchedule2 thursdaySchedule2 fridaySchedule2 saturdaySchedule2"
                            "moremondaySchedule moretuesdaySchedule morewednesdaySchedule morethursdaySchedule morefridaySchedule moresaturdaySchedule"
                            "linkAvability linkAvability linkAvability linkAvability linkAvability linkAvability"`,
      }}
    >
      {renderScheduleData()}
      {viewAllAvailabilityLinkUI({ onClickViewAllAvailability })}
    </Box>
  );
};
