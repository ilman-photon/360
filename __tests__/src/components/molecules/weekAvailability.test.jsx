import { fireEvent, render, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";
import { WeekAvailability } from "../../../../src/components/molecules/WeekAvailability/WeekAvailability";
import constants from "../../../../src/utils/constants";
const provider_data = [
  {
    date: "2022-09-19",
    list: [
      {
        time: "11:30am",
        key: 12222,
      },
    ],
  },
  {
    date: "2022-09-20",
    list: [
      {
        time: "08:00am",
        key: 12223,
      },
      {
        time: "10:30am",
        key: 12224,
      },
      {
        time: "11:00am",
        key: 12225,
      },
      {
        time: "12:00pm",
        key: 12226,
      },
      {
        time: "13:00pm",
        key: 12227,
      },
      {
        time: "14:00pm",
        key: 12228,
      },
    ],
  },
  {
    date: "2022-09-21",
    list: [
      {
        time: "08:30am",
        key: 12229,
      },
      {
        time: "10:30am",
        key: 12230,
      },
      {
        time: "11:30am",
        key: 12231,
      },
      {
        time: "12:00pm",
        key: 12232,
      },
      {
        time: "13:30pm",
        key: 12233,
      },
      {
        time: "14:30pm",
        key: 12234,
      },
      {
        time: "15:30pm",
        key: 12235,
      },
      {
        time: "16:30pm",
        key: 12236,
      },
      ,
    ],
  },
  {
    date: "2022-09-22",
    list: [
      {
        time: "09:30am",
        key: 12237,
      },
      {
        time: "11:00am",
        key: 12238,
      },
    ],
  },
  {
    date: "2022-09-23",
    list: [
      {
        time: "09:30am",
        key: 12239,
      },
    ],
  },
  {
    date: "2022-09-24",
    list: [
      {
        time: "09:30am",
        key: 12240,
      },
    ],
  },
]

describe("WeekAvailability Components", () => {
  let container;
  beforeEach(() => {
    container = render(
      <WeekAvailability onClickViewAllAvailability={jest.fn()} scheduleData={provider_data}/>
    );
  });

  it("WeekAvailability render", async () => {
    await waitFor(() => {
      container.getAllByText(/more/i);
    });
    expect(container.getAllByText(/more/i)[0]).toBeInTheDocument();
  });
});
