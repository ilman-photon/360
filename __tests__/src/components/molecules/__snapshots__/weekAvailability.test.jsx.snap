// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeekAvailability Components WeekAvailability render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-1jnq1v7"
      >
        <div
          class="buttonWrapper MuiBox-root css-129r4ni"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root scheduleBtn css-1ai4zn7-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 scheduleBtnLabel css-1jvxxc3-MuiTypography-root"
            >
              11:30am
            </p>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="dividerContainer MuiBox-root css-0"
        >
          <hr
            class="MuiDivider-root MuiDivider-fullWidth divider css-9mgopn-MuiDivider-root"
          />
        </div>
        <div
          class="dividerContainer MuiBox-root css-0"
        >
          <hr
            class="MuiDivider-root MuiDivider-fullWidth divider css-9mgopn-MuiDivider-root"
          />
        </div>
        <div
          class="dividerContainer MuiBox-root css-0"
        >
          <hr
            class="MuiDivider-root MuiDivider-fullWidth divider css-9mgopn-MuiDivider-root"
          />
        </div>
        <div
          class="buttonWrapper MuiBox-root css-1qlelg9"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root scheduleBtn css-1ai4zn7-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 scheduleBtnLabel css-1jvxxc3-MuiTypography-root"
            >
              08:00am
            </p>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="buttonWrapper MuiBox-root css-symg6n"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root scheduleBtn css-1ai4zn7-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 scheduleBtnLabel css-1jvxxc3-MuiTypography-root"
            >
              10:30am
            </p>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="buttonWrapper MuiBox-root css-1mdw407"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root scheduleBtn css-1ai4zn7-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 scheduleBtnLabel css-1jvxxc3-MuiTypography-root"
            >
              11:00am
            </p>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="buttonWrapper MuiBox-root css-enxit9"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root scheduleBtn css-1ai4zn7-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 scheduleBtnLabel css-1jvxxc3-MuiTypography-root"
            >
              3 more
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-puv47q-MuiSvgIcon-root"
                data-testid="KeyboardArrowDownOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                />
              </svg>
            </p>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="buttonWrapper MuiBox-root css-nvlpp0"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root scheduleBtn css-1ai4zn7-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 scheduleBtnLabel css-1jvxxc3-MuiTypography-root"
            >
              08:30am
            </p>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="buttonWrapper MuiBox-root css-eugzp0"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root scheduleBtn css-1ai4zn7-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 scheduleBtnLabel css-1jvxxc3-MuiTypography-root"
            >
              10:30am
            </p>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="buttonWrapper MuiBox-root css-7tv070"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root scheduleBtn css-1ai4zn7-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 scheduleBtnLabel css-1jvxxc3-MuiTypography-root"
            >
              11:30am
            </p>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="buttonWrapper MuiBox-root css-y2y9jp"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root scheduleBtn css-1ai4zn7-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 scheduleBtnLabel css-1jvxxc3-MuiTypography-root"
            >
              5 more
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-puv47q-MuiSvgIcon-root"
                data-testid="KeyboardArrowDownOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                />
              </svg>
            </p>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="buttonWrapper MuiBox-root css-1miqoiq"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root scheduleBtn css-1ai4zn7-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 scheduleBtnLabel css-1jvxxc3-MuiTypography-root"
            >
              09:30am
            </p>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="buttonWrapper MuiBox-root css-1ra7543"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root scheduleBtn css-1ai4zn7-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 scheduleBtnLabel css-1jvxxc3-MuiTypography-root"
            >
              11:00am
            </p>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="dividerContainer MuiBox-root css-0"
        >
          <hr
            class="MuiDivider-root MuiDivider-fullWidth divider css-9mgopn-MuiDivider-root"
          />
        </div>
        <div
          class="dividerContainer MuiBox-root css-0"
        >
          <hr
            class="MuiDivider-root MuiDivider-fullWidth divider css-9mgopn-MuiDivider-root"
          />
        </div>
        <div
          class="buttonWrapper MuiBox-root css-1t64eod"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root scheduleBtn css-1ai4zn7-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 scheduleBtnLabel css-1jvxxc3-MuiTypography-root"
            >
              09:30am
            </p>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="dividerContainer MuiBox-root css-0"
        >
          <hr
            class="MuiDivider-root MuiDivider-fullWidth divider css-9mgopn-MuiDivider-root"
          />
        </div>
        <div
          class="dividerContainer MuiBox-root css-0"
        >
          <hr
            class="MuiDivider-root MuiDivider-fullWidth divider css-9mgopn-MuiDivider-root"
          />
        </div>
        <div
          class="dividerContainer MuiBox-root css-0"
        >
          <hr
            class="MuiDivider-root MuiDivider-fullWidth divider css-9mgopn-MuiDivider-root"
          />
        </div>
        <div
          class="buttonWrapper MuiBox-root css-1kizmox"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root scheduleBtn css-1ai4zn7-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 scheduleBtnLabel css-1jvxxc3-MuiTypography-root"
            >
              09:30am
            </p>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="dividerContainer MuiBox-root css-0"
        >
          <hr
            class="MuiDivider-root MuiDivider-fullWidth divider css-9mgopn-MuiDivider-root"
          />
        </div>
        <div
          class="dividerContainer MuiBox-root css-0"
        >
          <hr
            class="MuiDivider-root MuiDivider-fullWidth divider css-9mgopn-MuiDivider-root"
          />
        </div>
        <div
          class="dividerContainer MuiBox-root css-0"
        >
          <hr
            class="MuiDivider-root MuiDivider-fullWidth divider css-9mgopn-MuiDivider-root"
          />
        </div>
        <div
          class="linkWrapper MuiBox-root css-wdc560"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways linkAvailabelity css-1ps4owl-MuiTypography-root-MuiLink-root"
          >
            View all Availability
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-1jnq1v7"
    >
      <div
        class="buttonWrapper MuiBox-root css-129r4ni"
      >
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root scheduleBtn css-1ai4zn7-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 scheduleBtnLabel css-1jvxxc3-MuiTypography-root"
          >
            11:30am
          </p>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="dividerContainer MuiBox-root css-0"
      >
        <hr
          class="MuiDivider-root MuiDivider-fullWidth divider css-9mgopn-MuiDivider-root"
        />
      </div>
      <div
        class="dividerContainer MuiBox-root css-0"
      >
        <hr
          class="MuiDivider-root MuiDivider-fullWidth divider css-9mgopn-MuiDivider-root"
        />
      </div>
      <div
        class="dividerContainer MuiBox-root css-0"
      >
        <hr
          class="MuiDivider-root MuiDivider-fullWidth divider css-9mgopn-MuiDivider-root"
        />
      </div>
      <div
        class="buttonWrapper MuiBox-root css-1qlelg9"
      >
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root scheduleBtn css-1ai4zn7-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 scheduleBtnLabel css-1jvxxc3-MuiTypography-root"
          >
            08:00am
          </p>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="buttonWrapper MuiBox-root css-symg6n"
      >
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root scheduleBtn css-1ai4zn7-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 scheduleBtnLabel css-1jvxxc3-MuiTypography-root"
          >
            10:30am
          </p>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="buttonWrapper MuiBox-root css-1mdw407"
      >
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root scheduleBtn css-1ai4zn7-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 scheduleBtnLabel css-1jvxxc3-MuiTypography-root"
          >
            11:00am
          </p>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="buttonWrapper MuiBox-root css-enxit9"
      >
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root scheduleBtn css-1ai4zn7-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 scheduleBtnLabel css-1jvxxc3-MuiTypography-root"
          >
            3 more
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-puv47q-MuiSvgIcon-root"
              data-testid="KeyboardArrowDownOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
              />
            </svg>
          </p>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="buttonWrapper MuiBox-root css-nvlpp0"
      >
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root scheduleBtn css-1ai4zn7-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 scheduleBtnLabel css-1jvxxc3-MuiTypography-root"
          >
            08:30am
          </p>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="buttonWrapper MuiBox-root css-eugzp0"
      >
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root scheduleBtn css-1ai4zn7-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 scheduleBtnLabel css-1jvxxc3-MuiTypography-root"
          >
            10:30am
          </p>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="buttonWrapper MuiBox-root css-7tv070"
      >
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root scheduleBtn css-1ai4zn7-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 scheduleBtnLabel css-1jvxxc3-MuiTypography-root"
          >
            11:30am
          </p>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="buttonWrapper MuiBox-root css-y2y9jp"
      >
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root scheduleBtn css-1ai4zn7-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 scheduleBtnLabel css-1jvxxc3-MuiTypography-root"
          >
            5 more
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-puv47q-MuiSvgIcon-root"
              data-testid="KeyboardArrowDownOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
              />
            </svg>
          </p>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="buttonWrapper MuiBox-root css-1miqoiq"
      >
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root scheduleBtn css-1ai4zn7-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 scheduleBtnLabel css-1jvxxc3-MuiTypography-root"
          >
            09:30am
          </p>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="buttonWrapper MuiBox-root css-1ra7543"
      >
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root scheduleBtn css-1ai4zn7-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 scheduleBtnLabel css-1jvxxc3-MuiTypography-root"
          >
            11:00am
          </p>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="dividerContainer MuiBox-root css-0"
      >
        <hr
          class="MuiDivider-root MuiDivider-fullWidth divider css-9mgopn-MuiDivider-root"
        />
      </div>
      <div
        class="dividerContainer MuiBox-root css-0"
      >
        <hr
          class="MuiDivider-root MuiDivider-fullWidth divider css-9mgopn-MuiDivider-root"
        />
      </div>
      <div
        class="buttonWrapper MuiBox-root css-1t64eod"
      >
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root scheduleBtn css-1ai4zn7-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 scheduleBtnLabel css-1jvxxc3-MuiTypography-root"
          >
            09:30am
          </p>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="dividerContainer MuiBox-root css-0"
      >
        <hr
          class="MuiDivider-root MuiDivider-fullWidth divider css-9mgopn-MuiDivider-root"
        />
      </div>
      <div
        class="dividerContainer MuiBox-root css-0"
      >
        <hr
          class="MuiDivider-root MuiDivider-fullWidth divider css-9mgopn-MuiDivider-root"
        />
      </div>
      <div
        class="dividerContainer MuiBox-root css-0"
      >
        <hr
          class="MuiDivider-root MuiDivider-fullWidth divider css-9mgopn-MuiDivider-root"
        />
      </div>
      <div
        class="buttonWrapper MuiBox-root css-1kizmox"
      >
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root scheduleBtn css-1ai4zn7-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 scheduleBtnLabel css-1jvxxc3-MuiTypography-root"
          >
            09:30am
          </p>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="dividerContainer MuiBox-root css-0"
      >
        <hr
          class="MuiDivider-root MuiDivider-fullWidth divider css-9mgopn-MuiDivider-root"
        />
      </div>
      <div
        class="dividerContainer MuiBox-root css-0"
      >
        <hr
          class="MuiDivider-root MuiDivider-fullWidth divider css-9mgopn-MuiDivider-root"
        />
      </div>
      <div
        class="dividerContainer MuiBox-root css-0"
      >
        <hr
          class="MuiDivider-root MuiDivider-fullWidth divider css-9mgopn-MuiDivider-root"
        />
      </div>
      <div
        class="linkWrapper MuiBox-root css-wdc560"
      >
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways linkAvailabelity css-1ps4owl-MuiTypography-root-MuiLink-root"
        >
          View all Availability
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
